doc/manual/src/architecture/architecture.md

```diff
@@ -0,0 +1,84 @@
+# Architecture
+
+This chapter describes how Nix works.
+It should help users understand why Nix behaves as it does, and it should help developers understand how to modify Nix and how to write similar tools.
+
+## Overview
+
+Nix consists of [hierarchical layers].
+
+[hierarchical layers]: https://en.m.wikipedia.org/wiki/Multitier_architecture#Layers
+
+```
++---------------------------------------------------------------+
+| Nix             .-------------------------.                   |
+|                 | commmand line interface |------.            |
+|                 '-------------------------'      |            |
+|                              |                   |            |
+|                          evaluates               |            |
+|                              |                manages         |
+|                              V                   |            |
+|                 .-------------------------.      |            |
+|                 | configuration language  |      |            |
+|                 '-------------------------'      |            |
+|                              |                   |            |
+|                         evaluates to             |            |
+|                              |                   V            |
+| +----------------------------|------------------------------+ |
+| | store                      |                              | |
+| |            referenced by   V       builds                 | |
+| | .-------------.      .------------.      .--------------. | |
+| | | build input |----->| build plan | ---->| build result | | |
+| | '-------------'      '------------'      '--------------' | |
+| +-----------------------------------------------------------+ |
++---------------------------------------------------------------+
```

Comment:
Not sure if I'm misunderstanding, but `referenced by` seems to go in the opposite direction of the arrow.

Reply:
Build inputs are referenced by build tasks.

Yes, the arrow directions can be ambiguous. As we discussed in yesterday's maintainer meeting, the point of these diagrams is to convey the right general idea in order to help shape the first approximation of a mental
model, not to be mathematically precise.

There is not enough space in such a diagram and not enough capacity on the reader's end to capture all the details, and this introductory document is not meant to hold that information.

The necessary details will follow follow-up PRs adding sub-chapters, which will be linked from the respective terms.

Reply:
Can’t we just invert the arrow and say “build inputs reference build tasks”?

Reply:
Alright, just wanted to double check. Consider my question answered. Thanks!